src/main/scala/edu/ie3/simona/agent/grid/GridAgent.scala

```diff
@@ -191,14 +192,21 @@ object GridAgent extends DBFSAlgorithm with DCMAlgorithm {
         SimonaActorNaming.actorName(ctx.self),
       )
 
-      constantData.resolution.foreach { resolution =>
-        constantData.environmentRefs.scheduler ! ScheduleActivation(
+      val resolution = constantData.resolution
+
+      val messageToScheduler = if (resolution == Long.MaxValue) {
+        unlockKey.unlock()
+        Completion(ctx.self)
```

Comment:
I think this doesn't work, since we have not been activated before

Reply:
Do you mean the unlock or the message?

Reply:
The message. The unlock is still necessary

Reply:
Ok. I will adapt this part.